### PR TITLE
Feature/upn iaea

### PR DESCRIPTION
**Pull request by whaeck**
Daniel Lopez Aldama (IAEA/NDS) provided a number of private fixes to NJOY2016:
- adding the ismooth input parameter to the acer input
- a fix concerning the proper use of the a parameter for Kalbach-Mann (MF6)
- a correction of the mt number for absorption competition for the probability tables

In addition to these provided fixes, the following is also included in this pull request
- adding the ismooth input parameter to the groupr input (similar to acer)
- adding tests 28 and 29 for testing ismooth options in acer and groupr
- adding test 30 for matxsr using both neutrons and photons

Reviewer still needs to be assigned.

### CODE REVIEW

File: src/groupr.f90 [hunk 1]
```diff
@@ -6826,7 +6818,14 @@ real(kr) function f6ddx(ep,epnext,epmax,w,cnow,lang,lep)
          endif
          ! kalbach-86 (obsolete kalbach-mann coding has been deleted)
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/groupr.f90]: No that comment line was already in the code since NJOY2012.

[the remark this answers is not in the document]
Comment (jlconlin) [reply src/groupr.f90]: I realize the comment has been around for awhile. I'm just asking if it is related to `na.eq.2`. If that code has been intentionally deleted, why? And if so, why should we be adding it back in?

[the remark this answers is not in the document]
Comment (whaeck) [reply src/groupr.f90]: I checked the code in 97 and 99, and this part of the code for aa is the same as well (no change compared to 2016). So whatever this comment is referring to has nothing to do with the case na=2.

**Comment by kahlerac** [comment]
Wim, Jeremy,
   I believe in the early days of Kalbach-Mann there was more than one way of doing things.  See 2012's ident up51 (or compare groupr2012.50 with groupr2012.51) to see what I changed.

**Comment by whaeck** [comment]
Thanks Skip, that is good to now.

By the way, I have asked around and it would appear that no ENDF/B evaluations actually use na=2 as an option (I have not checked the files myself so don't quote me on that). If that is true, the actual fix should have no impact on ENDF/B-VIII processing.

**Pull request merged**